EventProc/src/HcalDigiProducer.cxx

```diff
@@ -65,8 +69,14 @@ namespace ldmx {
 
         int numHCalSimHits = hcalHits->GetEntries();
         for (int iHit = 0; iHit < numHCalSimHits; iHit++) {
+            
             SimCalorimeterHit* simHit = (SimCalorimeterHit*) hcalHits->At(iHit);
             int detIDraw = simHit->getID();
+            detID_->setRawValue(detIDraw);
+            detID_->unpack();
+            int layer = detID_->getFieldValue("layer");
+            int subsection = detID_->getFieldValue("section");
+            int strip = detID_->getFieldValue("strip");                 
             std::vector<float> position = simHit->getPosition();
 	    
             //if we aggregate by layer, set all strip to zero and rcalculate the detIDraw
```

Comment:
Do we want to remove this? We won't do full layer anymore, and somebody might accidentally use that option if in there.

Reply:
I don't think there is an issue with keeping it as it retains backward compatibility